Mathlib/RingTheory/Etale/Locus.lean

```diff
@@ -0,0 +1,103 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+module
+
+public import Mathlib.RingTheory.Smooth.Locus
+public import Mathlib.RingTheory.Unramified.Locus
+
+/-!
+# Etale locus of an algebra
+
+## Main results
+- `Algebra.etaleLocus` : The set of primes that is etale over the base.
+- `Algebra.basicOpen_subset_etaleLocus_iff` :
+  `D(f)` is contained in the etale locus if and only if `A_f` is etale over `R`.
+- `Algebra.etaleLocus_eq_univ_iff` :
+  The etale locus is the whole spectrum if and only if `A` is etale over `R`.
```

Comment:
Would it make sense to add also a version of the lemma where we assume `A` is finitely presented, so that the RHS would involve `Etale` instead of `FormallyEtale`?
```suggestion
  The etale locus is the whole spectrum if and only if `A` is formally etale over `R`.
```